api/info.go

```diff
@@ -82,6 +82,10 @@ func NewInfoService(runner *Runner) *InfoService {
 // 	}
 //
 func (s *InfoService) GetInfo(r *http.Request, args *InfoArgs, requestBody *rpc.RequestBody, reply *InfoReply) error {
+	if s.runner.cfg.IsPublic {
```

Comment:
Can we not register unnecessary services?
If service has both admin and not admin methods - we can split it in two services

Reply:
Actually, I try to not register service twice but, we have `node` and `contract` which used in internal and public

Reply:
Services have to be registered once for different types of API. And there shouldn't be such checks it every method